code/src/components/GameElements.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+
+import  game, { continueGame }   from '../reducers/game'
+
+import "nes.css/css/nes.min.css";
+
+const GameElements = () => {
+  const gameElements = useSelector(store => store.game.gameElements)
+  const history = useSelector(store => store.game.history)
+  const userName = useSelector(store => store.game.userName) 
+
+  const dispatch = useDispatch()
+
+    return (
+      <div className="nes-container is-rounded is-dark game-container">
+        <p>{gameElements.description}</p> 
+          {/* // With gameElements.actions?.map we execute maping IF actions is an array. Shorthand to if statement. */}
+          {gameElements.actions.map(item => {
+            return ( 
+              <>
+                <p>{item.description}</p>
+                <button
+                  className="nes-btn is-error"
+                  key={item.direction}
+                  onClick={() => dispatch(continueGame(userName, item.direction))}
+                 >{item.direction}</button> 
+              </>   
+          )})}
+                <button onClick={() => dispatch(game.actions.setHistory())} disabled={!history.length}>Go back</button>
```

Comment:
The "play again" again button would be more accurate when the last level is reached 🤓